webpack/scenes/Subscriptions/Allocations/UpstreamSubscriptionsReducer.js

```diff
@@ -0,0 +1,32 @@
+import Immutable from 'seamless-immutable';
+
+import {
+  UPSTREAM_SUBSCRIPTIONS_REQUEST,
+  UPSTREAM_SUBSCRIPTIONS_SUCCESS,
+  UPSTREAM_SUBSCRIPTIONS_FAILURE,
+} from './UpstreamSubscriptionsContstants';
+
+const initialState = Immutable({ loading: true, results: [] });
+
+export default (state = initialState, action) => {
+  switch (action.type) {
+    case UPSTREAM_SUBSCRIPTIONS_REQUEST:
+      return state.set('loading', true);
+
+    case UPSTREAM_SUBSCRIPTIONS_SUCCESS:
+      return Immutable({
```

Comment:
Why not use `set` ?
something like:
```javascript
return state
  .set('result', action.response.results)
  .set(['loading', false])
  .set('searchIsActive', !!action.search)
```

Reply:
I can do that, don't really have a preference one way or the other.  Is there a benefit to doing it that way instead of like this?

Reply:
or you could use `merge`
```javascript
return state.merge({result: action.response.results, loading: false})
as far as I know, the performance is better this way.